client/src/Components/Tabs.js

```diff
@@ -2,60 +2,85 @@ import React, { useState, useEffect } from "react";
 import Pans from "./Pans";
 
 const Tabs = (props) => {
-    const [data, setData] = useState(props.data);
-    const todayDate = new Date().getDate();
-    const [isToday, setIsToday] = useState(true);
-    const [isTmr, setIsTmr] = useState(false);
-    const [isLater, setIsLater] = useState(false);
-    const todayData = data.filter(
-      (ele) => new Date(ele.by_date).getDate() === todayDate
-    );
-    const tmrData = data.filter(
-      (ele) => new Date(ele.by_date).getDate() === todayDate + 1
-    );
-    const laterData = data.filter(
-      (ele) =>
-        new Date(ele.by_date).getDate() !== todayDate + 1
-        && new Date(ele.by_date).getDate() !== todayDate
-    );
-    useEffect(() => {
-        setData(props.data);
-    }, [props.data]);
+  const [data, setData] = useState(props.data);
+  const todayDate = new Date().getDate();
+  const [isToday, setIsToday] = useState(true);
+  const [isTmr, setIsTmr] = useState(false);
+  const [isLater, setIsLater] = useState(false);
+  const [taskIsArchived, setIsArchived] = useState(false);
 
-  function handleClick(e, taskDate) {
+  const todayData = data.filter(
+    (ele) =>
+      new Date(ele.by_date).getDate() === todayDate
+      && ele.task_archived === false
+  );
+
+  const tmrData = data.filter(
+    (ele) =>
+      new Date(ele.by_date).getDate() === todayDate + 1
+      && ele.task_archived === false
+  );
+
+  const laterData = data.filter(
+    (ele) =>
+      new Date(ele.by_date).getDate() !== todayDate + 1
+      && new Date(ele.by_date).getDate() !== todayDate
+      && ele.task_archived === false
+  );
+  const archivedData = data.filter((ele) => ele.task_archived === true);
+
+  useEffect(() => {
+    setData(props.data);
+  }, [props.data]);
+
+  const handleClick = (e, taskDate) => {
     e.preventDefault();
     if (taskDate === "today") {
       setIsToday(true);
```

Comment:
You could look at using useReducer() to reduce the complexity here. But understanding and implementing useReducer is a challenge in itself. So don't worry about it if you can't quite understand how it would work.